### ISSUE TITLE
API endpoint for find/search

### ISSUE THREAD

**Issue by Ckath** [4chan/4chan-API#82]
Since there is no way to get 'all posts on all boards' through the api, or even 'all posts on this board', going over all posts programatically requires requests for the boards, then the threads on each board, then the posts in each thread. This is rather painful and annoying, also inefficient. (if there is a better way please do let me know)

The on site [search](https://boards.4channel.org/search) and this [rather hidden find page](https://find.4chan.org) would solve this issue, if they had an api endpoint. These pages arent ideal for programatically accessing due to their paging of results, which causes overlap when new posts are added while going through the pages.

tl;dr: API endpoint for either all posts, or a nice json response from the global search would be nice

**Comment by desuwa** [comment]
The API mostly maps to 4chan's HTML pages and, in general, we can't stray too far from this model.

To get all posts from a board, grab the [thread list](https://github.com/4chan/4chan-API/blob/master/pages/Threadlist.md) then fetch individual [threads](https://github.com/4chan/4chan-API/blob/master/pages/Threads.md).
Or you can use the catalog json too.

**Comment by Ckath** [comment]
That is what I currently settled on as well, it just results in a lot of api requests. If that's fine and it isnt going to change thats the end of this issue I guess.
